src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -140,6 +140,24 @@ internal static void ValidateScriptCompilationParameters(Compilation previousScr
             }
         }
 
+        internal static void CheckCompilationOptions(CompilationOptions options)
```

Comment:
I believe we check all options in the (CSharp|VB)CompilationOptions where we turn them into diagnostic instead of throwing an exception.